splinterd/src/main.rs

```diff
@@ -223,41 +230,35 @@ fn main() {
 
     let config = load_toml_config(config_file_path);
 
-    // Currently only YamlStorage is supported
-
     let node_id = matches
         .value_of("node_id")
         .map(String::from)
         .or_else(|| config.node_id())
-        .expect("Must provide a unique node id");
+        .ok_or_else(|| UserError::MissingArgument("node_id".into()))?;
 
     let storage_type = matches
         .value_of("storage")
         .map(String::from)
         .or_else(|| config.storage())
-        .or_else(|| Some(String::from("yaml")))
-        .expect("No Storage Provided");
+        .unwrap_or_else(|| String::from("yaml"));
```

Comment:
Can we add these defaults to the arg definitions in the clap app instead? That way they're documented in the CLI help as well.

Reply:
Shawn wants to remove all defaults from cli/clap.

Reply:
They will eventually be moved into a DefaultConfig struct.